libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshControllerTest.kt

```diff
@@ -715,71 +824,88 @@ class RouteRefreshControllerTest {
+                    routes = listOf(
+                        createTestTwoLegRoute(
+                            firstLegAnnotations = createRouteLegAnnotation(
+                                congestion = listOf("severe", "severe"),
+                                congestionNumeric = listOf(90, 90),
+                            )
+                        ),
+                        createTestTwoLegRoute(
+                            firstLegAnnotations = createRouteLegAnnotation(
+                                congestion = listOf("severe", "severe"),
+                                congestionNumeric = listOf(90, 90),
+                            )
                         )
                     )
                 )
             )
-        )
-        val routeRefreshStub = RouteRefreshStub().apply {
-            setRefreshedRoute(refreshedRoutes[0])
-            setRefreshedRoute(refreshedRoutes[1])
-        }
-        val refreshOptions = RouteRefreshOptions.Builder().build()
-        val routeRefreshController = createRouteRefreshController(
-            routeRefresh = routeRefreshStub,
-            routeRefreshOptions = refreshOptions
-        )
+            val routeRefreshStub = RouteRefreshStub().apply {
+                setRefreshedRoute(refreshedRoutes[0])
+                setRefreshedRoute(refreshedRoutes[1])
+            }
+            val refreshOptions = RouteRefreshOptions.Builder().build()
+            val routeRefreshController = createRouteRefreshController(
+                routeRefresh = routeRefreshStub,
+                routeRefreshOptions = refreshOptions
+            )
 
-        val refreshedRoutesDeferred = async {
-            routeRefreshController.refresh(initialRoutes)
-        }
-        advanceTimeBy(refreshOptions.intervalMillis)
+            val refreshedRoutesDeferred = async {
+                routeRefreshController.refresh(initialRoutes)
+            }
+            advanceTimeBy(refreshOptions.intervalMillis)
 
-        val result = refreshedRoutesDeferred.getCompletedTest()
+            val result = refreshedRoutesDeferred.getCompletedTest()
 
-        assertEquals(
-            RefreshedRouteInfo(refreshedRoutes, currentIndices),
-            result
-        )
-    }
+            assertEquals(
+                RefreshedRouteInfo(refreshedRoutes, currentIndices),
+                result
+            )
+            verify(ordering = Ordering.ORDERED) {
+                mockStatesObserver.onNewState(
+                    RouteRefreshStateResult(RouteRefreshExtra.REFRESH_STATE_STARTED)
+                )
+                mockStatesObserver.onNewState(
+                    RouteRefreshStateResult(RouteRefreshExtra.REFRESH_STATE_FINISHED_SUCCESS)
+                )
+            }
+        }
 
     @Test
-    fun `primary route refresh failed, alternative route refreshed successfully`() =
+    fun `primary route refresh failed, alternative route refreshed successfully, observer started, success`() =
```

Comment:
Could you update the SUCCESS state docs to mention that refresh state can be successful even when only a subset of routes is refreshed?